### DISCUSSION TITLE
Mmiddleware size is 1 MB and your plan size limit is 1 MB.

### DISCUSSION THREAD

**Discussion by terijaki** [vercel/community#6990] [Help]
### Summary

Hey there
I'm getting the following error when trying to deploy my NextJS project to Vercel. Not really sure when this started. I'm only using AuthJS in my middleware. I cannot really understand the root of this.

> Error: The Edge Function "middleware" size is 1 MB and your plan size limit is 1 MB.

It is saying that the middleware is too big, but then the build log shows only 163 kB

![image](https://github.com/vercel/community/assets/590522/38097a8a-92db-4532-84bc-5b10a4d29269)

I used the Next Bundle Analyzer which also shows values below 1MB.
![image](https://github.com/vercel/community/assets/590522/b950da71-e8f3-4b84-89cf-f0606781c026)

Is there any known issue on Vercel? Anything I can do to troubleshoot besides the bundle analyzer?

Thank you ✌️

### Steps to Reproduce

1. Setup NextJS auth AuthJS
2. Configure AuthJS with Prisma Adapter
3. Deploy to Vercel (Hobby)

**Comment by TomAndril** [comment]
Hi, I was facing the same issue when I migrated to AuthJS v5. Followed the steps under "Edge Compatibility" on [this link](https://authjs.dev/getting-started/migrating-to-v5#edge-compatibility) and now works again! Hope it helps

**Reply by terijaki** [reply]
Thanks. I am using prisma though and they added edge compatibility (https://authjs.dev/getting-started/adapters/prisma).

It’s not the compatibility I struggle with but rather the sudden 1MB Middleware, despite the logs claiming it’s half the size.

**Reply by alexcjwei** [reply]
I'm getting the same issue, but with the `/api/auth/[...nextauth]` route:
```
Route (app)                              Size     First Load JS
┌ ƒ /                                    141 B          87.2 kB
├ ƒ /_not-found                          871 B          87.9 kB
├ ƒ /api/auth/[...nextauth]              0 B                0 B
├ ƒ /api/chat                            0 B                0 B
├ ƒ /api/embeddings/files/[id]           0 B                0 B
├ ƒ /api/files                           0 B                0 B
├ ƒ /games                               6.95 kB          94 kB
├ ƒ /games/[id]                          1.08 kB        88.1 kB
└ ƒ /landing                             141 B          87.2 kB
+ First Load JS shared by all            87 kB
  ├ chunks/23-cbbd9e3af8bf5d6a.js        31.5 kB
  ├ chunks/fd9d1056-be48aeae6e94b8d1.js  53.6 kB
  └ other shared chunks (total)          1.86 kB
  ```

The build error is:
```
Error: The Edge Function "api/auth/[...nextauth]" size is 1 MB and your plan size limit is 1 MB. Learn More: https://vercel.link/edge-function-size
```

I'm also using the Prisma/authjs setup with edge compatibility. Here's the route:
```
import { handlers } from '@/lib/auth';

export const { GET, POST } = handlers;

export const runtime = 'edge';
```

The issue is similar to above where the size is 0B, but somehow the route is over 1MB

**Reply by UygarDeniz** [reply]
> Thanks. I am using prisma though and they added edge compatibility (https://authjs.dev/getting-started/adapters/prisma).
> 
> It’s not the compatibility I struggle with but rather the sudden 1MB Middleware, despite the logs claiming it’s half the size.

It's probably too late to answer but I had the same problem. I deleted prisma edge compatibility codes and separated auth-config just like how they say in their docs. It solved my problem.

**Comment by EarlGeorge** [comment]
upgrade till latest sdk and it will work

[a comment by an automated account is not shown]
